include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h

```diff
@@ -16,12 +16,15 @@
 #ifndef _ONEDPL_PARALLEL_BACKEND_SYCL_MERGE_SORT_H
 #define _ONEDPL_PARALLEL_BACKEND_SYCL_MERGE_SORT_H
 
-#include <cmath>     // std::log2
-#include <limits>    // std::numeric_limits
-#include <cassert>   // assert
-#include <utility>   // std::swap
-#include <cstdint>   // std::uint32_t, ...
-#include <algorithm> // std::min
+#include <bit>         // std::bit_floor
+#include <cmath>       // std::log2
+#include <limits>      // std::numeric_limits
+#include <cassert>     // assert
+#include <utility>     // std::swap
+#include <cstdint>     // std::uint32_t, ...
+#include <variant>     // std::variant, std::visit
```

Comment:
@dmitriy-sobolev please take a look to https://github.com/oneapi-src/oneDPL/pull/1755 - I have implemented required logic without variant - visitor usage.

Reply:
Thanks for the idea of the strategy. I see it as introduction of 2 layers, where each one serves its one purpose: dispatching basing on an index or a leaf type. This is one more proof of "We can solve any problem by introducing an extra level of indirection" theorem :)
 
I've adopted it [0daae6e](https://github.com/oneapi-src/oneDPL/pull/1735/commits/0daae6e5e218f458c47d11c89600900673ea70de) with some modifications.